src/main/java/org/mqttbee/mqtt/handler/MqttChannelInitializer.java

```diff
@@ -67,6 +113,40 @@ protected void initChannel(final SocketChannel channel) {
 
         pipeline.addLast(MqttConnectHandler.NAME, new MqttConnectHandler(connect, connAckEmitter, clientData));
         pipeline.addLast(MqttDisconnectHandler.NAME, channelComponent.disconnectHandler());
+
     }
 
+    private void initMqttOverWebSockets(@NotNull final ChannelPipeline pipeline) {
+        try {
+            final URI serverUri = new URI(getScheme(), null,
+                    clientData.getServerHost(), clientData.getServerPort(), "/" + clientData.getServerPath(), null, null);
+            final MqttWebSocketClientProtocolHandler wsProtocolHandler = new MqttWebSocketClientProtocolHandler(serverUri);
+
+            wsProtocolHandler.onWebSocketHandshakeComplete(() -> {
+                pipeline.addLast(WS_BINARYFRAME_CREATOR, new WebSocketBinaryFrameEncoder());
+                pipeline.addLast(WS_BINARYFRAME_UNPACKER, new WebSocketBinaryFrameDecoder());
+                this.initMqttHandlers(pipeline);
+            });
+
+            pipeline.addLast(HTTP_CODEC, new HttpClientCodec());
+            pipeline.addLast(HTTP_AGGREGATOR, new HttpObjectAggregator(MAXIMUM_PACKET_SIZE_LIMIT));
+            pipeline.addLast(WEBSOCKETS_CLIENT, wsProtocolHandler);
+
+        } catch (URISyntaxException e) {
+            Exceptions.propagate(e);
+        }
+    }
+
+    private String getScheme() {
+        return clientData.usesSSL() ? "was" : "ws";
+    }
+
+    private void init(@NotNull final Channel channel) {
+        channelComponent = ChannelComponent.create(channel, clientData);
```

Comment:
Why is here a second ChannelComponent created?

Reply:
was from a previously fixed bug. Removed.